felix/fv/donottrack_test.go

```diff
@@ -38,6 +39,196 @@ import (
 	"github.com/projectcalico/calico/libcalico-go/lib/options"
 )
 
+var _ = infrastructure.DatastoreDescribe("_BPF-SAFE_ do-not-track policy tests; with bond", []apiconfig.DatastoreType{apiconfig.Kubernetes}, func(getInfra infrastructure.InfraFactory) {
+	if !BPFMode() {
+		return
+	}
+	var (
+		infra          infrastructure.DatastoreInfra
+		tc             infrastructure.TopologyContainers
+		hostW          [2]*workload.Workload
+		client         client.Interface
+		cc             *Checker
+		externalClient *containers.Container
+	)
+
+	BeforeEach(func() {
+		var err error
+		infra = getInfra()
+		options := infrastructure.DefaultTopologyOptions()
+		if BPFMode() {
+			options.ExtraEnvVars["FELIX_BPFLogLevel"] = "debug"
+			options.IPIPEnabled = false
+		}
+		tc, client = infrastructure.StartNNodeTopology(2, options, infra)
+		cc = &Checker{}
+
+		// Start a host networked workload on each host for connectivity checks.
+		for ii := range tc.Felixes {
+			// We tell each workload to open:
+			// - its normal (uninteresting) port, 8055
+			// - port 2379, which is both an inbound and an outbound failsafe port
+			// - port 22, which is an inbound failsafe port.
+			// This allows us to test the interaction between do-not-track policy and failsafe
+			// ports.
+			const portsToOpen = "8055,2379,22"
+			hostW[ii] = workload.Run(
+				tc.Felixes[ii],
+				fmt.Sprintf("host%d", ii),
+				"default",
+				tc.Felixes[ii].IP, // Same IP as felix means "run in the host's namespace"
+				portsToOpen, "tcp")
+		}
+
+		// We will use this container to model an external client trying to connect into
+		// workloads on a host.  Create a route in the container for the workload CIDR.
+		externalClient = infrastructure.RunExtClient("ext-client")
+		err = infra.AddDefaultDeny()
+		Expect(err).To(BeNil())
+	})
+
+	JustAfterEach(func() {
+		if CurrentGinkgoTestDescription().Failed {
+			for _, felix := range tc.Felixes {
+				if NFTMode() {
+					logNFTDiags(felix)
+				} else {
+					felix.Exec("iptables-save", "-c")
+					felix.Exec("ip6tables-save", "-c")
+				}
+				felix.Exec("ip", "r")
+				felix.Exec("calico-bpf", "policy", "dump", "eth0", "all", "--asm")
+				felix.Exec("calico-bpf", "-6", "policy", "dump", "eth0", "all", "--asm")
+				felix.Exec("calico-bpf", "counters", "dump")
+			}
+		}
+	})
+
+	AfterEach(func() {
+		tc.Stop()
+		infra.Stop()
+		externalClient.Stop()
+	})
+
+	Context("after adding eth0 to a bond", func() {
```

Comment:
woudn't it be easier to create  `Context("with bond", ...)` inside the current Describe to save all this copy-pasta and just do the BeforeEach to setup the bond?